--- libmodbus-3.0.3/src/modbus-rtu.c	2012-05-20 01:54:15.000000000 +0400
+++ libmodbus-3.0.3-indigo/src/modbus-rtu.c	2012-09-28 14:38:03.942124167 +0400
@@ -723,6 +723,13 @@
         modbus_rtu_t *ctx_rtu = ctx->backend_data;
         struct serial_rs485 rs485conf;
         memset(&rs485conf, 0x0, sizeof(struct serial_rs485));
+        rs485conf.flags |= SER_RS485_RTS_ON_SEND;
+        rs485conf.flags &= ~(SER_RS485_RTS_AFTER_SEND);
+        rs485conf.delay_rts_before_send = 15;
+        rs485conf.delay_rts_after_send = 15;
+
+        rs485conf.flags |= SER_RS485_RX_DURING_TX;
+
 
         if (mode == MODBUS_RTU_RS485) {
             rs485conf.flags = SER_RS485_ENABLED;
