diff -ruN dbus-1.2.26/bus/config-parser-trivial.c dbus-1.2.26-my/bus/config-parser-trivial.c
--- dbus-1.2.26/bus/config-parser-trivial.c	2011-02-20 16:40:37.000000000 +0300
+++ dbus-1.2.26-my/bus/config-parser-trivial.c	2011-02-20 16:47:15.000000000 +0300
@@ -329,14 +329,6 @@
   return &parser->service_dirs;
 }
 
-dbus_bool_t
-bus_config_parser_check_doctype (BusConfigParser   *parser,
-                                 const char        *doctype,
-                                 DBusError         *error)
-{
-    return TRUE;
-}
-
 #ifdef DBUS_BUILD_TESTS
 #include <stdio.h>
 #include "test.h"
